### PR TITLE
feat(TSE-848): Add support for the new In-Context Editor

### PR DESCRIPTION
**Pull request by Varpusparvi**
BREAKING CHANGES

This PR replaces the old ICE with the new ICE. This integration was using the ruby api client, which was now removed since the new ICE doesn't need it. Readme was also rewritten as well as old packages upgraded/removed in case they were not needed anymore. Users can still use the old ICE via 1.x.x versions of this gem.

- Remove support for old In-Context Editor
  - Remove API client related code since it's not used anymore
  - Replace old asset link with new ICE link
  - Remove old config options not in use with new ICE
- Add support for the new In-Context Editor
  - Replace old asset link with new ICE link
  - Adjust old config to better match what new ICE supports 
- Rewrote readme
- Restyled with `standard`
- Upgraded ruby version
- Update packages, removed unused packages
- Update CODEOWNERS
- Remove Zendesk integration

https://phrase.atlassian.net/browse/TSE-848

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,19 +36,19 @@ Next, create the initializer file by executing the Rails generator:
...
+You can find the ID of your account in your account settings in Translation Center.
+
 ### Add the JavaScript helper
 
 Next, add the Javascript helper to your Rails application layout file:
 
-    <%= phraseapp_in_context_editor_js %>
+<%= phraseapp_in_context_editor_js %>
 
 If you don't want to use the helper but add the plain Javascript yourself, head over to our [documentation](https://help.phrase.com/help/translate-directly-on-your-website) to learn more.
```
Comment (itsahsiao) [inline README.md:53]: This link https://help.phrase.com/help/translate-directly-on-your-website that's used throughout the README goes to the CLI 🤔 

Maybe we should link to https://support.phrase.com/hc/en-us/articles/5784095916188-In-Context-Editor-Strings- instead?

The link for the In-Context Editor Demo under References section is also outdated.

File: lib/generators/phraseapp_in_context_editor/templates/phraseapp_in_context_editor.rb [hunk 2]
```diff
@@ -5,23 +5,16 @@
   # Fetch your project id after creating your first project
   # in Translation Center.
   # You can find the project id in your project settings
-  # page (https://phraseapp.com/projects)
+  # page (https://app.phrase.com/)
```
Comment (itsahsiao) [inline lib/generators/phraseapp_in_context_editor/templates/phraseapp_in_context_editor.rb:8]:
```suggestion
  # Configure with your project id and account id. You can find the 
  # project id in your project settings and account id in your account
  # settings (https://app.phrase.com/)
```

**Comment by itsahsiao** [comment]
@jablan @theSoenke @anthonyshchang could one of you support with reviewing this PR or testing that this version of the Ruby integration works for Rails? 🙏

**Comment by Varpusparvi** [comment]
https://github.com/phrase/phraseapp-in-context-editor-ruby/pull/57
Here is the PR for added demo, feel free to try it out. It's based on this PR so the integration will work the same

**Comment by jablan** [comment]
@itsahsiao can someone add some closer info in the PR description on what the change is about? "add support for new ICE" doesn't mean much without knowing, at least roughly, what has been changed and which kind of changes it required in ruby gem.

File: lib/phraseapp-in-context-editor-ruby/view_helpers.rb [hunk 3]
```diff
@@ -1,35 +1,41 @@
...
-          </script>
-        eos
+        snippet = <<-EOS
+        <script>
+          window.PHRASEAPP_CONFIG = #{configuration.to_json};
+          (function() {
+            let phraseapp = document.createElement('script');
+            phraseapp.type = 'module';
+            phraseapp.async = true;
+            phraseapp.src = "https://d2bgdldl6xit7z.cloudfront.net/latest/ice/index.js";
```
Comment (jablan) [inline lib/phraseapp-in-context-editor-ruby/view_helpers.rb:30]: is this URL something that never changes? looks pretty random to me.

Comment (Varpusparvi) [reply lib/phraseapp-in-context-editor-ruby/view_helpers.rb:30]: It never changes unless editor will be deployed to a different place 👍

Comment (theSoenke) [reply lib/phraseapp-in-context-editor-ruby/view_helpers.rb:30]: @Varpusparvi i think we had a chat about the URL before. maybe we can create a task for the devops team to provider a better URL on the phrase domain?

Comment (Varpusparvi) [reply lib/phraseapp-in-context-editor-ruby/view_helpers.rb:30]: We can also do that and we can then change these in other places also fyi @itsahsiao

File: README.md [hunk 4]
```diff
@@ -1,107 +1,122 @@
-# Phrase In-Context Editor for Ruby #
+# phraseapp-in-context-editor-ruby
 
-*This Ruby gem is for use with Ruby (Rails, Sinatra) applications only. Check out the [documentation](https://help.phrase.com/help/translate-directly-on-your-website) to learn how to set up the In-Context Editor with other technologies.*
+![Build status](https://github.com/phrase/phraseapp-in-context-editor-ruby/workflows/Test/badge.svg)
 
-![Build Status](https://github.com/phrase/phraseapp-in-context-editor-ruby/workflows/Test/badge.svg)
+**phraseapp-in-context-editor-ruby** is the official library for integrating [Phrase Strings In-Context Editor](https://support.phrase.com/hc/en-us/articles/5784095916188-In-Context-Editor-Strings) with [i18n](https://github.com/ruby-i18n/i18n) in your React application.
```
Comment (itsahsiao) [inline README.md:5]: Bad copy/paste? 😁  It should say `Ruby (Rails, Sinatra)` application, instead of `React` at the end of the sentence.

**Review by jablan — APPROVED** [review]

File: .gitignore [hunk 5]
```diff
@@ -3,7 +3,6 @@
 .bundle
 .config
 .yardoc
-Gemfile.lock
```
Comment (theSoenke) [inline .gitignore:6]: For a library I think it's a general best practice to not commit the lockfile but rather just have some constraints in the gemspec

**Comment by theSoenke** [comment]
@Varpusparvi maybe a separate task but should we also rename this gem? `phraseapp` is a little outdated 😬 probably would also require publishing a new package to rubygems as gems cannot directly be renamed

File: .gitignore (hunk 5 above)
Comment (Varpusparvi) [reply .gitignore:6]: I'd like to keep it as without it if someone clones our repo they might install different versions of dev environment which might cause problems if they want to improve this gem.

**Pull request merged**